app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -134,39 +127,13 @@ fun ConversationScreen(
         conversationInfoViewModel.observeConversationDetails()
     }
 
-    with(conversationCallViewModel) {
-        if (conversationCallViewState.shouldShowJoinAnywayDialog) {
-            appLogger.i("showing showJoinAnywayDialog..")
-            JoinAnywayDialog(
-                onDismiss = ::dismissJoinCallAnywayDialog,
-                onConfirm = ::joinAnyway
-            )
-        }
+    val dialogStateType = remember { mutableStateOf(ConversationScreenDialogType.NONE) }
```

Comment:
With that, we have two places where we keep the information whether a specific dialog should be shown: `dialogStateType` and in ViewModels, for instance `shouldShowJoinAnywayDialog`.
I think we should decide if we want to keep this information in ViewModels or treat it as UI specific thing and do the same we did for snackbars - create a Flow which is being collected in Composable and the ViewModel doesn't keep this information, only informs UI that it needs to be shown.